contracts/service-registry/src/contract.rs

```diff
@@ -255,9 +256,19 @@ pub mod execute {
             .may_load(deps.storage, (&service_name, &info.sender))?
             .ok_or(ContractError::WorkerNotFound)?;
 
+        let mut worker_current_chains = CHAINS_PER_WORKER
+            .may_load(deps.storage, (&service_name, &info.sender))?
+            .unwrap_or_else(HashSet::new);
+
         for chain in chains {
             WORKERS_PER_CHAIN.save(deps.storage, (&service_name, &chain, &info.sender), &())?;
+            worker_current_chains.insert(chain.clone());
         }
+        CHAINS_PER_WORKER.save(
+            deps.storage,
+            (&service_name, &info.sender),
+            &(worker_current_chains),
+        )?;
```

Comment:
I think you can just use update, instead of load and then store.

Reply:
then we should move the code inside the loop, which i believe will severely affect performance. Here, we are only doing load/save pair once. But using update will call it with every loop.

Reply:
to can provide the `update` function with an action instead of a value. So no need to put it into the loop, you can append the chains inside the action.